buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt

```diff
@@ -227,27 +229,13 @@ fun ComponentMetadataHandler.withLibraryDependencies(module: String, kClass: KCl
 }
 
 
-open class DowngradeIvyRule : ComponentMetadataRule {
+open class DowngradeOpentest4jRule : ComponentMetadataRule {
     override fun execute(context: ComponentMetadataContext) {
         context.details.allVariants {
-            withDependencyConstraints {
-                filter { it.group == "org.apache.ivy" }.forEach {
-                    it.version { prefer("2.2.0") }
-                    it.because("Gradle depends on ivy implementation details which changed with newer versions")
-                }
-            }
-        }
-    }
-}
-
-
-open class DowngradeTestNGRule : ComponentMetadataRule {
-    override fun execute(context: ComponentMetadataContext) {
-        context.details.allVariants {
-            withDependencyConstraints {
-                filter { it.group == "org.testng" }.forEach {
-                    it.version { prefer("6.3.1") }
-                    it.because("6.3.1 is required by Gradle and part of the distribution")
+            withDependencies {
+                filter { it.group == "org.opentest4j" }.forEach {
+                    it.version { strictly("1.0.0") }
```

Comment:
@marcphilipp Do we now have an `opentest4j` version where this problem is fixed?

Reply:
Yes, and I'll update to it in #6728.